Lib/test/test_thread.py

```diff
@@ -265,5 +267,107 @@ def tearDown(self):
             pass
 
 
+class TestStdioAtForkReInit(unittest.TestCase):
```

Comment:
I dislike such function test, it's fragile and may break tomorrow. I would prefer to remove it.

If you want to keep it, you should:

* create a new script and run the script from the test. So you better control the Python state when the test is run: which modules are imported, signals, etc.
* move it to test_io: the tested code lives in the _io module

Reply:
Can you clarify why you think it's fragile? It seems useful as a regression test. I do agree that it would be more stable in a separate script.

Reply:
This is the only part I didn't change yet, b/c I wanted to check with you guys on the details.

@vstinner All the at_fork_reinit handlers in the codebase have your name on them, so I'll trust your judgment here. (Though I'm also curious for the same reasons @JelleZijlstra mentioned).

On the topic of the current test, you said "If you want to keep it, you should..."

What's the "else" condition for that sentence? I would prefer to do whatever you think makes the most sense. I don't have any strong opinions here, aside from a desire to have that one line in there that reinitializes the lock.

Test-wise, would you prefer (1) just moving the current test to a separate script? (2) removing the current test entirely? (3) a different test with certain properties that this one doesn't have?

I'm equally happy to do any of the above. :)